pkg/download/dependency_resolution/resolver/basic_dep_resolver.go

```diff
@@ -75,9 +77,9 @@ func basicFindAndReplaceIDs(apiName string, configToBeUpdated config.Config, con
 // in case two configs are actually the same, or if they are both dashboards no replacement will happen as in these
 // cases there is no real valid reference even if the key is found in the content.
 func shouldReplaceReference(configToBeUpdated config.Config, configToUpdateFrom config.Config, contentToBeUpdated, keyToReplace string) bool {
-	if configToBeUpdated.Coordinate.Type == "dashboard" && configToUpdateFrom.Coordinate.Type == "dashboard" {
-		return false //dashboards can not actually reference each other, but often contain a link to another inside a markdown tile
+	if !canReference(configToBeUpdated, configToUpdateFrom) {
+		return false
 	}
 
-	return configToUpdateFrom.Template.ID() != configToBeUpdated.Template.ID() && strings.Contains(contentToBeUpdated, keyToReplace)
+	return strings.Contains(contentToBeUpdated, keyToReplace)
```

Comment:
To any reviewer: Does someone know why we compared the template-id here? It should be the coordinate like in the other resolver, right? 

@UnseenWizzard could you maybe also have a look?

Reply:
Is there a case where two configs are downloaded with the same template file? OR is this a crude way to ensure that references are not created to a config itself?

Reply:
Would it be valid to have both checks?
- A config should not reference itself, but
- Two different configs with the same template (can that occur?) cannot reference each other.

Reply:
The issue I have is that the aho-algo does compare the coordinates; only the basic did not do that. Since all tests work fine without adoption, I think we're fine. I just want to double-check it.

Just triple-checked it, the Template-ID is usually initialized using the remove config-id of the config. Since this is part of the coordinate, it should be absolutely fine. We should have it consistent though.